http/src/main/java/io/micronaut/http/util/CookieUtil.java

```diff
@@ -0,0 +1,210 @@
+                .toFormatter();
+
+        private final String cookieName;
+        private final String cookieValue;
+        private final Map<String, String> props;
+
+        public HeaderPropertyMapCookieConfiguration(String cookieName, String cookieValue, Map<String, String> props) {
+            this.cookieName = cookieName;
+            this.cookieValue = cookieValue;
+            this.props = props;
+        }
+
+        public Cookie toCookie() {
+            return new SimpleCookie(cookieName, cookieValue)
+                    .configure(this, isCookieSecure().orElse(false));
+        }
+
+        @Override
+        @NonNull
+        public String getCookieName() {
+            return cookieName;
+        }
+
+        @Override
+        public Optional<String> getCookieDomain() {
+            return Optional.ofNullable(props.get(CookieProperties.DOMAIN.toLowerCase()));
+        }
+
+        @Override
+        public Optional<String> getCookiePath() {
+            return Optional.ofNullable(props.get(CookieProperties.PATH.toLowerCase()));
+        }
+
+        @Override
+        public Optional<Boolean> isCookieHttpOnly() {
+            return Optional.of(props.containsKey(CookieProperties.HTTP_ONLY.toLowerCase()));
+        }
+
+        public Optional<Boolean> isCookieSecure() {
+            return Optional.of(props.containsKey(CookieProperties.SECURE.toLowerCase()));
+        }
+
+        @Override
+        public Optional<TemporalAmount> getCookieMaxAge() {
+            // https://tools.ietf.org/html/rfc6265#page-19 - Max-Age has precedence over Expires.
+            if (props.get(CookieProperties.MAX_AGE.toLowerCase()) != null) {
+                String propertyValue = props.get(CookieProperties.MAX_AGE.toLowerCase());
+                long maxAge = propertyValue.isEmpty() ? 0 : Long.parseLong(propertyValue);
+                return Optional.ofNullable(Duration.ofSeconds(maxAge));
+            } else if (props.get(CookieProperties.EXPIRES.toLowerCase()) != null) {
+                Optional<LocalDateTime> expiry = parseCookieDate(props.get(CookieProperties.EXPIRES.toLowerCase()));
+                return expiry.map($ -> Duration.between($, LocalDateTime.now()));
+            } else {
+                return Optional.empty();
+            }
+        }
+
+        private Optional<LocalDateTime> parseCookieDate(String propertyValue) {
+
+            Optional<LocalDateTime> localDateTime = Arrays.stream(new DateTimeFormatter[]{
+                    DateTimeFormatter.RFC_1123_DATE_TIME,
+                    RFC_5322_RFC_850_FORMATTER,
+                    ANSI_C_ASCTIME_FORMATTER
+            }).map(dateTimeFormatter -> {
+                try {
+                    return LocalDateTime.parse(propertyValue, dateTimeFormatter);
+                } catch (DateTimeParseException e) {
+                    //ignored, try other patterns
+                    return null;
+                }
+            }).filter(Objects::nonNull).findFirst();
```

Comment:
Seems a bit of unnecessary overhead to use streams here. Would it not be enough to simply do:

```java
DateTimeFormatter daateTimeFormatter = new DateTimeFormatter[]{
                    DateTimeFormatter.RFC_1123_DATE_TIME,
                    RFC_5322_RFC_850_FORMATTER,
                    ANSI_C_ASCTIME_FORMATTER
}
try {
    return Optional.of(LocalDateTime.parse(propertyValue, dateTimeFormatter));
} catch (DateTimeParseException e) {
    return Optional.empty();
}
```